src/world/action/jump.c

```diff
@@ -1,15 +1,250 @@
+            phi_a0 = 0x10008;
+          }
+
+          temp_s1 = phi_a0;
+          suggest_player_anim_clearUnkFlag(temp_s1);
+          cam = &gCameras;
+          cam->moveFlags |= 1;
+        }
+    } else {
+        return;
+    }
+    playerStatus->fallState++;
+}
+
+void func_802B6294_E242D4() {
+    s32 temp_v1;
+    s32 phi_a0;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam;
+
+    if ((playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS)) {
+        func_802B647C_E244BC();
+        return;
+    }
+
+    temp_v1 = playerStatus->flags;
+    if ( playerStatus->flags < 0) {
+        playerStatus->flags &= 0x7FFFFFF5;
+        playerStatus->flags |= 4;
+
+        if (playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO) {
+            phi_a0 = 0x90005;
+        } else {
+            temp_v1 = 0x6000A;
+            phi_a0 = temp_v1;
+
+            if (!(playerStatus->animFlags & PLAYER_ANIM_FLAG_3)) {
+                phi_a0 = 0x10008;
+            }
+        }
+        suggest_player_anim_clearUnkFlag(phi_a0);
+        cam = &gCameras;
+        cam->moveFlags |= 1;
+    }
+    playerStatus->decorationList++;
+}
+
+void func_802B6348_E24388() {
+    f32* sub_arg1;
+    f32* sub_arg2;
+    f32* sub_arg3;
+    f32* sub_arg4;
+    f32 sub_arg5;
+    f32 sub_arg6;
+    f32 sub_arg7;
+    f32 sub_arg8;
+    f32 sp28;
+    f32 sp2C;
+    f32 sp30;
+    f32 sp34;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam;
+
+    if (playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS) {
+        func_802B6508_E24548();
+        return;
+    }
+
+    if (playerStatus->flags < 0) {
+        playerStatus->flags &= 0x7FFFFFF5;
+        playerStatus->flags |= 4;
+        cam = &gCameras;
+        cam->moveFlags |= 1;
+    }
+
+    playerStatus->decorationList++;
+    phys_update_interact_collider();
+    sub_arg1 = &sp28;
+    sub_arg2 = &sp2C;
+    sub_arg3 = &sp30;
+    sub_arg4 = &sp34;
+    sub_arg6 = sub_arg6;
+    sp28 = playerStatus->position.x;
+    sp2C = playerStatus->position.y;
+    sp30 = playerStatus->position.z;
+    sp34 = playerStatus->colliderHeight;
+
+    if (((get_collider_type_by_id(player_raycast_below_cam_relative(playerStatus, sub_arg1, sub_arg2, sub_arg3, sub_arg4,
+                                    &sub_arg5, &sub_arg6, &sub_arg7, &sub_arg8)) & 0xFF) - 2 >= 2u) && check_input_jump()) {
+
+        set_action_state(3);
+        playerStatus->flags &= -0xF;
+        func_802B60B4_E240F4();
+    }
+}
+
+void func_802B647C_E244BC() {
+    s32 temp_v1;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam;
+
+    if (playerStatus->flags < 0) {
+        playerStatus->flags &= 0x7FFFFFFF;
+        playerStatus->flags &= ~0xA;
+        playerStatus->flags |= 4;
+        temp_v1 = 0xA0006;
+
+        if (!(playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS)) {
+            temp_v1 = 0x10008;
+        }
+        suggest_player_anim_clearUnkFlag(temp_v1);
```

Comment:
```suggestion
        if (playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS) {
            suggest_player_anim_clearUnkFlag(0xA0006);
        } else {
            suggest_player_anim_clearUnkFlag(0x10008);
       }
```

GCC likes to break up multiple of the same function call in different branches like this. It just pulls out the arguments and then calls the function at the end. Try play around with the branching here and you can get rid of a bunch of these temps you're using, this applies to all the `suggest_player_anim_clearUnkFlag` here, and probably more functions besides.